src/util/logging/loggingWrapper.js

```diff
@@ -0,0 +1,74 @@
+import ky from 'ky';
+
+const logBuffer = [];
+const MAX_LOGS = 20;
+const FLUSH_INTERVAL = 5000;
+
+const logDedupMap = new Map();
+const DEDUP_WINDOW_MS = 10000; // 10 seconds
+
+const shouldLog = message => {
+	const now = Date.now();
+
+	if (logDedupMap.has(message)) {
+		const lastLogged = logDedupMap.get(message);
+
+		if (now - lastLogged < DEDUP_WINDOW_MS) {
+			return false;
+		}
+	}
+
+	logDedupMap.set(message, now);
+
+	return true;
+};
+
+const flushLogs = async () => {
+	if (!logBuffer.length) return;
+	const logsToSend = [...logBuffer];
+	logBuffer.length = 0;
+
+	ky.post('http://localhost:5000/logs', {
```

Comment:
Take a look here: https://github.com/yaptide/ui/blob/95a1e3b098a9f3cbf4e07e7ec55181fb7c3aea50/src/config/ConfigService.tsx#L7

backend URL shouldn;t be hardcoded.